### PR TITLE
Default Cassandra Read Body Error Message

### PR DESCRIPTION
**Pull request by msindwan**
We recently encountered an error where the translation was not found for a particular cassandra error. We speculated that this was a write failure and subsequently added it to the list of error codes as per the [cassandra protocol](https://git-wip-us.apache.org/repos/asf?p=cassandra.git;a=blob;f=doc/native_protocol_v5.spec;hb=HEAD). In addition, we added a safe guard for error translations to gracefully handle unsupported error codes.

### CODE REVIEW

**Comment by thibaultcha** [comment]
Oh nice! It seems like `WRITE_FAILURE` was added in binary protocol v4 (https://github.com/apache/cassandra/blob/trunk/doc/native_protocol_v4.spec#L1106) - while this library was originally developed against the v3. Thanks for the catch!

I am currently in the process of improving the CQL implementation quite a lot (before improving the high-level interfaces). I believe we are having some issues with CI due to third-party dependencies, and those will have to be resolved before I can merge this. Will try to do so as soon as possible!

Thanks again!

File: lib/cassandra/cql.lua [hunk 1]
```diff
@@ -1377,7 +1380,16 @@ do
     elseif op_code == OP_CODES.ERROR then
       local code = body:read_int()
       local message = body:read_string()
-      return nil, '['..ERROR_TRANSLATIONS[code]..'] '..message, code
+      local error_translation = ERROR_TRANSLATIONS[code]
+
+      -- If the translation is not found, return a formatted string
+      -- with the error code for convenience.
+      if error_translation == nil then
+        error_translation = string.format(
```
Comment (thibaultcha) [inline lib/cassandra/cql.lua]: We should be using the localized version of `string.format` (in the `fmt` local) instead here.

File: lib/cassandra/cql.lua [hunk 2]
```diff
@@ -1377,7 +1380,16 @@ do
...
+          'UNSUPPORTED ERROR (code=%d)', code or ERRORS.UNKNOWN)
```
Comment (thibaultcha) [inline lib/cassandra/cql.lua]: I don't think we should use the `%d` modifier here - using `fmt("0x%x", code)` will print the error code in the same format the binary protocol specification refers to it.

File: lib/cassandra/cql.lua [hunk 3]
```diff
@@ -102,12 +102,14 @@ local ERRORS            = {
...
   READ_FAILURE          = 0x1300,
   FUNCTION_FAILURE      = 0x1400,
+  WRITE_FAILURE         = 0x1500,
   SYNTAX_ERROR          = 0x2000,
   UNAUTHORIZED          = 0x2100,
   INVALID               = 0x2200,
   CONFIG_ERROR          = 0x2300,
   ALREADY_EXISTS        = 0x2400,
   UNPREPARED            = 0x2500,
+  UNKNOWN               = -1,
```
Comment (thibaultcha) [inline lib/cassandra/cql.lua]: I don't think this constant is necessary. The binary protocol guarantees us we will be receiving an error code already.

Comment (msindwan) [reply lib/cassandra/cql.lua]: Nice! Will remove then

**Comment by thibaultcha** [comment]
Merged, thank you!

**Pull request merged**